plugins/bundle/plugin.go

```diff
@@ -429,3 +509,82 @@ func (p *Plugin) configDelta(newConfig *Config) (map[string]*Source, map[string]
 
 	return newBundles, updatedBundles, deletedBundles
 }
+
+func (p *Plugin) saveBundleToDisk(name string, b *bundle.Bundle) error {
+
+	bundleDir := filepath.Join(p.bundlePersistPath, name)
+
+	if _, err := os.Stat(filepath.Join(bundleDir, "bundle.tar.gz")); err == nil {
+		err := saveCurrentBundleToDisk(bundleDir, ".bundle.tar.gz.tmp", b)
+		if err != nil {
+			p.logWarn(name, "Failed to save new bundle to disk: %v", err)
+			err = os.Remove(filepath.Join(bundleDir, ".bundle.tar.gz.tmp"))
+			if err != nil {
+				p.logWarn(name, "Failed to remove temp file ('.bundle.tar.gz.tmp'): %v", err)
+			}
+			return nil
+		}
+	} else {
+		err := saveCurrentBundleToDisk(bundleDir, ".bundle.tar.gz.tmp", b)
+		if err != nil {
+			rErr := os.Remove(filepath.Join(bundleDir, ".bundle.tar.gz.tmp"))
+			if rErr != nil {
+				p.logWarn(name, "Failed to remove temp file ('.bundle.tar.gz.tmp'): %v", rErr)
+			}
+			return err
+		}
+	}
```

Comment:
nit: Can refactor this to reduce duplicate code like:

```go
	tmpFile := filepath.Join(bundleDir, ".bundle.tar.gz.tmp")
	bundleFile := filepath.Join(bundleDir, "bundle.tar.gz")

	saveErr := saveCurrentBundleToDisk(bundleDir, ".bundle.tar.gz.tmp", b)
	if saveErr != nil {
	    p.logWarn(name, "Failed to save new bundle to disk: %v", saveErr)

		if err := os.Remove(tmpFile); err != nil {
			p.logWarn(name, "Failed to remove temp file ('%s'): %v", tmpFile, err)
		}

		if _, err := os.Stat(bundleFile); err == nil {
		    p.logWarn(name, "Older version of activated bundle persisted, ignoring error")
			return nil
		}
		return saveErr
	}
	
	return os.Rename(tmpFile, bundleFile)
```
(note there is a log message when we ignore the error too)

Reply:
Refactored the code as per your suggestion.